# 5. AspC simulation of heterologous expression

In [48]:
%run 1_Objective_function.ipynb

ASPtex ASPtex: asp__L_e <=> asp__L_p L-aspartate transport via diffusion (extracellular to periplasm)
EX_asp__L_e EX_asp__L_e: asp__L_e -->  L-Aspartate exchange
0.87699721425716
0.8769972144269869


_Pseudomonas aeruginosa_'s AspC has a 30% higher kcat than E. coli's AspC for L-aspartate production. Therefore, we will change the lower and upper bounds of the kcat of AspC to 30% higher than the default.

In [49]:
original_solution = model.optimize()
print("Original Objective Value for Biomass:", original_solution.objective_value)

Original Objective Value for Biomass: 0.8769972144269869


In [50]:
reaction_id = "ASPTA"
# Check if the reaction is in the model
if reaction_id in [rxn.id for rxn in model.reactions]:
    # Get the reaction object
    reaction = model.reactions.get_by_id(reaction_id)
    print(f"Reaction {reaction_id} is present in the model.")
    print("Reaction details:", reaction)
    print(f"Lower bound: {reaction.lower_bound}")
    print(f"Upper bound: {reaction.upper_bound}")

Reaction ASPTA is present in the model.
Reaction details: ASPTA: akg_c + asp__L_c <=> glu__L_c + oaa_c
Lower bound: -1000.0
Upper bound: 1000.0


In [51]:
model.objective = model.reactions.EX_asp__L_e
model.optimize().objective_value

4.596027557488632

In [52]:
# Get the reaction object
aspta_reaction = model.reactions.get_by_id(reaction_id)

# Modify the bounds by multiplying them by 1.3
aspta_reaction.lower_bound *= 1.3
aspta_reaction.upper_bound *= 1.3

# Print the modified bounds
print(f"New lower bound for {reaction_id}: {aspta_reaction.lower_bound}")
print(f"New upper bound for {reaction_id}: {aspta_reaction.upper_bound}")


New lower bound for ASPTA: -1300.0
New upper bound for ASPTA: 1300.0


In [53]:
model.objective = model.reactions.EX_asp__L_e
model.optimize().objective_value

4.5960275574885125

There is almost no change in max aspartate production (our objective value) before and after adjusting the bounds, (4.596027557488632 initially and 4.5960275574885125 after the change), so we will not change the bounds of the ASPTA reaction.